### PR TITLE
More doc test

### PR DESCRIPTION
**Pull request by jojolepro**
## PR Checklist

By placing an x in the boxes I certify that I have:

- [X] Ran `cargo test --all` locally if this modified any rs files.
- [X] Ran `cargo +stable fmt --all` locally if this modified any rs files.
- [X] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

### CODE REVIEW

File: amethyst_utils/src/circular_buffer.rs [hunk 1]
```diff
@@ -2,23 +2,40 @@
...
             queue: VecDeque::with_capacity(size),
+            cap: size,
         }
     }
 
     ///Add a value to the CircularBuffer
     ///Returns the popped value if the buffer is full
     pub fn push(&mut self, elem: A) -> Option<A> {
-        let out = if self.queue.len() == self.queue.capacity() {
+        let out = if self.queue.len() == self.cap {
```
Comment (azriel91) [inline amethyst_utils/src/circular_buffer.rs]: the queue capacity is dynamic right? (it grows when the buffer fills up)

Comment (Frizi) [reply amethyst_utils/src/circular_buffer.rs]: The thing is that `queue.capacity()` is unreliable. It is guaranteed to be AT LEAST what you specified, but can be greater than it. https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=dc6c7019a244ca8853df62a5728563de

**Comment by torkleyy** [comment]
Jenkins failed due to incorrect formatting (`cargo fmt` should make this PR pass).

**Comment by torkleyy** [comment]
:shipit:
bors r+

**Pull request merged**